### DISCUSSION TITLE
Use PGVectorStore in a more restrictive db environment, no "auto-setup"

### DISCUSSION THREAD

**Discussion by matzutt** [run-llama/LlamaIndexTS#1284] [Q&A] [answered]
Hi there!

We are currently migrating our llamaindex python service to typescript and make use of the PGVectorStore. We face the following problem:
For our production and development environment we use a hosted postgresql where no user has the privillege to create a schema.
Unfortunately the PGVectorStore tries to check the schema availability with a "CREATE SCHEMA IF NOT EXISTS..." which will fail due to missing access rights. The python llamaindex PGVectorStore had the option `perform_setup=false` to skip schema and table generation, so it was working there. For production systems, it can generally be a good idea to set up databases "by hand".

Because the checkSchema method is private I cannot override it, so I would need to replace the complete class.
Any better ideas or should I propose a simple change on the code and PR?

Thanks for your input!

**Comment by matzutt** [comment]
Would also be nice to accept a pg.Pool directly and not just the client - the pool itself has the method query and distributes queries across available clients or fires up new clients.

**Comment by marcusschiesser** [comment]
@matzutt seems like we have `fromPG(client: pg.Client | pg.PoolClient)` which excepts a pool client. Does that work for you?

Yes, please send a PR for making the schema check optional

**Reply by matzutt** [reply]
Regarding the pg.PoolClient - yes that helps and we used that before, but the pool is more flexible otherwise the VectorStore that is quite heavily used has only one client. But the "no schema check" is more important. If you like, I can propose a pool solution in another PR, to not mix up the features.

**Comment by matzutt** [comment, marked as answer]
I read the contribution guidelines, but I am still not sure, what the preferred way to a PR is. Should I convert this discussion into an issue and then create a branch for that issue (that is how we do it in our projects). Are new tests mandatory? Sorry for those newbie questions.
